awx/main/tasks/jobs.py

```diff
@@ -669,15 +505,15 @@ def run(self, pk, **kwargs):
             self.build_extra_vars_file(self.instance, private_data_dir)
             args = self.build_args(self.instance, private_data_dir, passwords)
             env = self.build_env(self.instance, private_data_dir, private_data_files=private_data_files)
-            self.safe_env = build_safe_env(env)
+            self.runner_callback.safe_env = build_safe_env(env)
 
             credentials = self.build_credentials_list(self.instance)
 
             for credential in credentials:
                 if credential:
                     credential.credential_type.inject_credential(credential, env, self.safe_cred_env, args, private_data_dir)
 
-            self.safe_env.update(self.safe_cred_env)
+            self.runner_callback.safe_env.update(self.safe_cred_env)
```

Comment:
You have a few things that need to be copied over from the task class to the callback class, but they're spread out throughout the `run` method. It would make the code easier to read if you consolidated them. Here would work. Those assignments look to be:
 - instance
 - parent_workflow_job_id
 - job_created
 - safe_env (assignment and then an update)

Just putting them all in 1 place would be nice. Maybe that will suggest that a helper method would help, but maybe not.

Reply:
I have made the changes , will be pushed in the next commit.